src/dataloader.py

```diff
@@ -0,0 +1,100 @@
+                                      if member.name.startswith(folder_inside)]
+                    zip_ref.extractall(path=destination_path, members=folder_members)
+                    #Relocate the images to the destination_path
+                    for member in folder_members:
+                        if os.path.dirname(member.name):
+                            os.rename(os.path.join(destination_path,member.name),
+                                      os.path.join(destination_path,os.path.basename(member.name)))
+                    #Delete empty folder
+                    DeleteFolder = os.path.split(Path(folder_inside))
+                    shutil.rmtree(os.path.join("data/images",DeleteFolder[0]))
+                    print("Unzipped in..." + destination_path)
+
+            except Exception as e:
+                print("Error extracting from ZIP file:", e)
+
+        # Unzipp a folder from inside a TAR/TGZ file to a destination
+        elif file_path.lower().endswith('.tar') or file_path.lower().endswith('.tgz'):
+            try:
+                with tarfile.open(file_path, 'r:gz') as tar_ref:
+                    print("Unzipping folder..." + folder_inside)
+                    all_members = tar_ref.getmembers()
+                    folder_members = [member for member in all_members 
+                                      if member.name.startswith(folder_inside)]
+                    tar_ref.extractall(path=destination_path, members=folder_members)
+                    #Relocate the images to the destination_path
+                    for member in folder_members:
+                        if os.path.dirname(member.name):
+                            os.rename(os.path.join(destination_path,member.name),
+                                      os.path.join(destination_path,os.path.basename(member.name)))
+                    #Delete empty folder
+                    DeleteFolder = os.path.split(Path(folder_inside))
+                    shutil.rmtree(os.path.join("data/images",DeleteFolder[0]))
+                    print("Unzipped in..." + destination_path)
+
+            except Exception as e:
+                print("Error extracting from TAR/TGZ file:", e)
+
+def list_files_in_folder(folder_path):
+    try:
+        # Get a list of all files and directories in the folder
+        files_in_folder = os.listdir(folder_path)
+        # Show only jpg files
+        files = [file for file in files_in_folder if file.lower().endswith('.jpg')]
+        return files
+    except Exception as e:
+        print("Error listing files in folder:", e)
+        return None
+    
+class MyDataset(data.Dataset):
+    # Call the super().__init__() method in the __init__.
+    def __init__(self, images_path, mask_path, transform=None):
+        super().__init__()
+        
+        self.images_path = images_path
+        self.mask_path = mask_path
+        self.transform = transform
+  
```

Comment:
Afegeix les imatges ordenades com a metode
`        self.image_files = sorted([file for file in os.listdir(images_path) if file.lower().endswith('.jpg')])`